lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -1027,13 +1238,60 @@ abstract class DefaultTextEditingStrategy implements TextEditingStrategy {
     assert(isEnabled);
 
     final EditingState newEditingState = EditingState.fromDomElement(activeDomElement);
+    TextEditingDeltaState? newTextEditingDeltaState;
+    if (inputConfiguration.enableDeltaModel) {
+      newTextEditingDeltaState = TextEditingDeltaState.inferDeltaState(newEditingState, lastEditingState, editingDeltaState);
+    }
 
     if (newEditingState != lastEditingState) {
       lastEditingState = newEditingState;
-      onChange!(lastEditingState);
+      _editingDeltaState = newTextEditingDeltaState;
+      onChange!(lastEditingState, _editingDeltaState);
+      // Flush delta after it has been sent to framework.
+      _editingDeltaState = null;
```

Comment:
This makes a lot more sense to me at a glance now, thanks for putting up with my pickiness.